File: 05 SWEA/D1/Java/src/Prob2056.java

/**
* title  : 2056. 연월일 달력 (D1)
* time   : 109ms
* memory : 19112kb
*/

import java.io.*;
 
public class Prob2056 {
    
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int[] day = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

        int T = Integer.parseInt(br.readLine());
        for (int t = 1; t <= T; t++) {

            String date = br.readLine();
            int y = Integer.parseInt(date.substring(0, 4));
            int m = Integer.parseInt(date.substring(4, 6));
            int d = Integer.parseInt(date.substring(6, 8));
            
            String answer = "-1";
            if (1 <= m && m <= 12 && 1 <= d && d <= day[m]) {
                answer = (String.format("%04d", y) + "/" 
                            + String.format("%02d", m) + "/" 
                            + String.format("%02d", d));
            } 

            sb.append("#" + t + " ").append(answer).append("\n");
        }

        System.out.println(sb);
    }
}

File: 05 SWEA/D1/Python/Prob2056.py

"""
title  : 2056. 연월일 달력 (D1)
time   : 101ms
memory : 43984kb
"""

def solution():

    maxDay = [0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

    T = int(input())
    for tc in range(1, T+1):

        str = input().rstrip()
        y, m, d = str[:4], str[4:6], str[6:]

        nm, nd = int(m), int(d)
        if (1 <= nm <= 12 and 1 <= nd <= maxDay[nm]):
            print(f'#{tc} {y}/{m}/{d}')
        else:
            print(f'#{tc} -1')

solution()